src/containerapp/azext_containerapp/custom.py

```diff
@@ -1135,25 +1135,29 @@ def update_managed_environment(cmd,
         logs_destination = None if logs_destination == "none" else logs_destination
         safe_set(env_def, "properties", "appLogsConfiguration", "destination", value=logs_destination)
 
-    if logs_destination == "log-analytics" and (not logs_customer_id or not logs_key):
-        raise ValidationError("Must provide logs-workspace-id and logs-workspace-key if updating logs destination to type 'log-analytics'.")
-
-    if logs_customer_id and logs_key:
+    if logs_destination == "log-analytics":
+        if not logs_customer_id or not logs_key:
+            raise ValidationError("Must provide logs-workspace-id and logs-workspace-key if updating logs destination to type 'log-analytics'.")
         safe_set(env_def, "properties", "appLogsConfiguration", "logAnalyticsConfiguration", "customerId", value=logs_customer_id)
         safe_set(env_def, "properties", "appLogsConfiguration", "logAnalyticsConfiguration", "sharedKey", value=logs_key)
 
+    # When using Azure Monitor Log Destination, Log Analytics can only be configured under Resource Diagnotic settings
+    if logs_destination == "azure-monitor":
+        safe_set(env_def, "properties", "appLogsConfiguration", "logAnalyticsConfiguration", value=None)
+
     # Custom domains
-    safe_set(env_def, "properties", "customDomainConfiguration", value={})
-    cert_def = env_def["properties"]["customDomainConfiguration"]
     if hostname:
+        if not certificate_file or not certificate_password:
+            raise ValidationError("Must provide certificate_file and certificate_password if updating the DNS suffix for the environment's custom domain.")
+        safe_set(env_def, "properties", "customDomainConfiguration", value={})
```

Comment:
why do you want to set a null value

Reply:
safe_set(env_def, "properties", "customDomainConfiguration", value={})
for init the "properties.customDomainConfiguration" path;

       `safe_set(env_def, "properties", "customDomainConfiguration", value={})
        cert_def = env_def["properties"]["customDomainConfiguration"]
        blob, _ = load_cert_file(certificate_file, certificate_password)
        safe_set(cert_def, "dnsSuffix", value=hostname)
        safe_set(cert_def, "certificatePassword", value=certificate_password)
        safe_set(cert_def, "certificateValue", value=blob)`
or

`       safe_set(env_def, "properties", "customDomainConfiguration", "dnsSuffix", value=hostname)
        safe_set(env_def, "properties", "customDomainConfiguration", "certificatePassword", value=certificate_password)
        safe_set(env_def, "properties", "customDomainConfiguration", "certificateValue", value=blob)`